app/src/main/java/com/rootstrap/android/ui/activity/main/targetpoint/MapFragment.kt

```diff
@@ -90,6 +100,48 @@ class MapFragment : PermissionFragment(), OnMapReadyCallback {
         setMapCamera(location)
     }
 
+    private fun addCircleMarkerForTarget(target: TargetModel) {
+        val position = LatLng(target.lat, target.lng)
+        target.topic?.run {
+            mMap.addMarker(
+                MarkerOptions()
+                    .position(position)
+                    .icon(bitmapDescriptorwithOvalBackground(requireContext(), getIconForTarget()))
+            )
+        }
+    }
+
+    private fun bitmapDescriptorwithOvalBackground(context: Context, @DrawableRes vectorDrawableResourceId: Int): BitmapDescriptor? {
```

Comment:
I think that this fun. should be async. because you are creating/manipulating a bitmap.

Reply:
also, you need to run this in the iOThread for better performance and to avoid blocking the main thread.

Reply:
perfect, I moved to a coroutine